src/confab/agent/init_test.go

```diff
@@ -42,7 +40,7 @@ type FakeAgentOutput struct {
 	PID  int
 }
 
-func getFakeAgentOutput(runner agent.Runner) FakeAgentOutput {
+func getFakeAgentOutput(runner *agent.Runner) FakeAgentOutput {
```

Comment:
Yes, some of the runner tests leak go routines ([for example](https://github.com/cloudfoundry-incubator/consul-release/blob/master/src/confab/agent/runner_test.go#L323), this leads to a panic (`sync: negative WaitGroup counter`) as the leaked go routine exits after the global `runner` variable has been reinitialized.  A panic here is the correct behavior as reusing a `Runner` is a programming error, doing otherwise risks leaking the underlying the `exec.Cmd`.

Swapping pointers, solves this problem and is the correct implementation.